python/tests/avro/test_reader.py

```diff
@@ -0,0 +1,162 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=W0212
+import pathlib
+
+from iceberg.avro.reader import AvroStructProtocol, DataFileReader
+from tests.io.test_io_base import LocalInputFile
+
+
+def test_read_header():
+    test_file_path = str(pathlib.Path(__file__).parent.resolve()) + "/manifest.avro"
```

Comment:
What do you think of adding a `tests/avro/data` directory, and then using a `pytest.parametrize` decorator for this test. We could then include a collection of manifest files and test different combinations of `filename`, `expected_header_magic`, `expected_header_meta`, and `expected_header_sync`.

We could do this for `test_reader_data(...)` as well.